src/hooks/useAdditionalAssetData.ts

```diff
@@ -1,118 +1,47 @@
-      coingeckoIds[address?.toLowerCase()]
-  );
-
-  // @ts-ignore
-  const coingeckoData: AdditionalDataCongecko | undefined = useSelector(
-    // @ts-ignore
-    ({ additionalAssetsData: { coingeckoData } }): AppState =>
-      coingeckoData[address?.toLowerCase()]
+): {
+  description?: string;
+  loading: boolean;
+  totalVolume?: string;
+  totalLiquidity?: string;
+  marketCap?: string;
+  links: Record<string, string[]>;
+} {
+  const address = rawAddress === ETH_ADDRESS ? WETH_ADDRESS : rawAddress;
+  const { data } = useQuery(['additionalAssetData', address], () =>
+    getAdditionalAssetData(address)
   );
-
-  // @ts-ignore
-  const uniswapData: AdditionalDataUniswap | undefined = useSelector(
-    // @ts-ignore
-    ({ additionalAssetsData: { uniswapData } }): AppState =>
-      uniswapData[address?.toLowerCase()]
-  );
-
-  const [totalLiquidity, setTotalLiqudity] = useState<number | undefined>();
-  const getTotalLiquidity = useCallback(async () => {
-    const id =
-      address?.toLowerCase() === 'eth' ? WETH_ADDRESS : address.toLowerCase();
-    const uniswapData = await getUniswapV2Tokens([id]);
-    const token = uniswapData?.[id];
-    setTotalLiqudity(token?.totalLiquidity);
-  }, [address]);
-  const dispatch = useDispatch();
-
   const { nativeCurrency } = useAccountSettings();
-
   const format = useCallback(
     value =>
       value
         ? bigNumberFormat(cutIfOver10000(value), nativeCurrency, value >= 10000)
         : '',
     [nativeCurrency]
   );
-
   const rate = useNativeCurrencyToUSD();
-
-  useEffect(() => {
-    coingeckoId &&
-      !coingeckoData &&
-      dispatch(additionalAssetsDataAddCoingecko(address?.toLowerCase()));
-  }, [coingeckoId, address, coingeckoData, dispatch]);
-
-  useEffect(() => {
-    !uniswapData &&
-      dispatch(additionalAssetsDataAddUniswap(address?.toLowerCase()));
-  }, [coingeckoId, address, coingeckoData, dispatch, uniswapData]);
-
-  useEffect(() => {
-    getTotalLiquidity();
-  }, [getTotalLiquidity]);
-
-  const newData = {
-    description: coingeckoData?.description,
-    totalVolumeLoading: typeof uniswapData?.oneDayVolumeUSD !== 'number',
-    ...(uniswapData?.oneDayVolumeUSD
-      ? {
-          totalVolume: format(uniswapData?.oneDayVolumeUSD * rate),
-        }
-      : {}),
-    marketCapLoading:
-      typeof coingeckoData?.circulatingSupply !== 'number' && !!coingeckoId,
-    ...(coingeckoData?.circulatingSupply
-      ? {
-          marketCap: format(coingeckoData?.circulatingSupply * tokenPrice),
-        }
-      : {}),
-    totalLiquidityLoading: typeof totalLiquidity !== 'number',
-    ...(totalLiquidity
-      ? {
-          totalLiquidity: format(tokenPrice * totalLiquidity),
-        }
-      : {}),
-    links: coingeckoData?.links,
+  const loading = !data;
+  return {
+    description: data?.description,
+    links: data?.links,
+    loading,
+    marketCap: format(data?.circulatingSupply * tokenPrice),
```

Comment:
I realize this is not from your changes, but could we make `format` and `cutIfOver10000` handle these calculations with our bignumber utilities (multiply, greaterThan, etc) instead of just dealing with numbers